machineconfiguration/v1/types_machineconfignode.go

```diff
@@ -158,6 +158,78 @@ type MachineConfigNodeStatus struct {
 	// +kubebuilder:validation:MaxItems=32
 	// +optional
 	IrreconcilableChanges []IrreconcilableChangeDiff `json:"irreconcilableChanges,omitempty"`
+	// internalReleaseImage describes the status of the release payloads stored in the node.
+	// +optional
+	InternalReleaseImage *MachineConfigNodeStatusInternalReleaseImage `json:"internalReleaseImage,omitempty"`
+}
+
+// MachineConfigNodeStatusInternalReleaseImage holds information about the current, desidered and discovered release bundles for the observed machine
+// config node.
+// +openshift:enable:FeatureGate=NoRegistryClusterOperations
+type MachineConfigNodeStatusInternalReleaseImage struct {
+	// conditions represent the observations of an internal release image current state.
+	// +listType=map
+	// +listMapKey=type
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=256
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// availableReleases is a list of release bundle identifiers currently detected
+	// from the ISO attached to one of the control plane nodes. Any reported identifier can
+	// be used to amend the `spec.Releases` field to add a new release bundle to the cluster.
+	// An empty value indicates that no ISOs are currently being detected on any control plane
+	// node.
+	// Must not exceed 5 entries.
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MaxItems=5
+	// +optional
```

Comment:
Just thinking out loud, maybe we should have this be required (if the IRI status exists) so at it would always reflect what the daemon is currently detecting? And if we don't detect anything we just have an empty list?